src/Sylius/Component/Core/Customer/Statistics/CustomerStatisticsProvider.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Component\Core\Customer\Statistics;
+
+use Sylius\Component\Core\Model\CustomerInterface;
+use Sylius\Component\Core\Model\OrderInterface;
+use Sylius\Component\Core\Repository\OrderRepositoryInterface;
+use Sylius\Component\Resource\Repository\RepositoryInterface;
+
+/**
+ * @author Jan Góralski <jan.goralski@lakion.com>
+ */
+final class CustomerStatisticsProvider implements CustomerStatisticsProviderInterface
+{
+    /**
+     * @var OrderRepositoryInterface
+     */
+    private $orderRepository;
+
+    /**
+     * @var RepositoryInterface
+     */
+    private $channelRepository;
+
+    /**
+     * @param OrderRepositoryInterface $orderRepository
+     * @param RepositoryInterface $channelRepository
+     */
+    public function __construct(OrderRepositoryInterface $orderRepository, RepositoryInterface $channelRepository)
+    {
+        $this->orderRepository = $orderRepository;
+        $this->channelRepository = $channelRepository;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getCustomerStatistics(CustomerInterface $customer)
+    {
+        $channels = $this->channelRepository->findAll() ?: [];
```

Comment:
Why do we need this? If we have all customer orders, then it should be enough to calculate statistics. It should reduce complicity also